Data/Graph.hs

```diff
@@ -255,34 +276,68 @@ type Bounds  = (Vertex, Vertex)
 -- | An edge from the first vertex to the second.
 type Edge    = (Vertex, Vertex)
 
--- | All vertices of a graph.
+-- | Returns the list of vertices in the graph.
+--
+-- ==== __Examples__
+--
+-- > vertices (buildG (0,-1) []) == []
+--
+-- > vertices (buildG (0,2) [(0,1),(1,2)]) == [0,1]
```

Comment:
Which vertices? If your second example is right, it's not all.

Reply:
Oops, I originally had the bounds incorrect `(0,1)` which just silently discarded the last vertex. I fixed it but forgot to update the output.

As a side note, this API is really hard to use properly.

Reply:
Yes, the API is very strange. It kind of looks like it wasn't meant for public consumption. My *guess* is that it may have started out GHC-internal, and someone else wanted to use it, and it ended up being exposed to the world before it really had a proper API. Those were fairly early days in Haskell history.

Reply:
I wonder if there's any reasonable way to figure out who's even using this API. It looks like there's some pretty low hanging fruit when it comes to improving the API (like hiding the functions you get back from `graphFromEdges`), but that would be a breaking API change. Maybe we could create a new module like `Data.DirectedGraph` which had the nice new API and just leave the old `Data.Graph` alone (module performance and comment improvements). WDYT?